crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +140,245 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
             }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment method".to_string()).into()),
+            PaymentMethodData::CardDetailsForNetworkTransactionId(_)
+            | PaymentMethodData::CardRedirect(_)
+            | PaymentMethodData::Wallet(_)
+            | PaymentMethodData::PayLater(_)
+            | PaymentMethodData::BankRedirect(_)
+            | PaymentMethodData::BankDebit(_)
+            | PaymentMethodData::BankTransfer(_)
+            | PaymentMethodData::Crypto(_)
+            | PaymentMethodData::MandatePayment
+            | PaymentMethodData::Reward
+            | PaymentMethodData::RealTimePayment(_)
+            | PaymentMethodData::MobilePayment(_)
+            | PaymentMethodData::Upi(_)
+            | PaymentMethodData::Voucher(_)
+            | PaymentMethodData::GiftCard(_)
+            | PaymentMethodData::OpenBanking(_)
+            | PaymentMethodData::CardToken(_)
+            | PaymentMethodData::NetworkToken(_) => Err(errors::ConnectorError::NotImplemented(
+                get_unimplemented_payment_method_error_message("jpmorgan"),
+            )
+            .into()),
         }
     }
 }
-
-//TODO: Fill the struct with respective fields
-// Auth Struct
+#[derive(Debug)]
 pub struct JpmorganAuthType {
-    pub(super) api_key: Secret<String>,
+    pub(super) _api_key: Secret<String>,
+    pub(super) _key1: Secret<String>,
 }
 
 impl TryFrom<&ConnectorAuthType> for JpmorganAuthType {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(auth_type: &ConnectorAuthType) -> Result<Self, Self::Error> {
         match auth_type {
-            ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
+            ConnectorAuthType::BodyKey { api_key, key1 } => Ok(Self {
+                _api_key: api_key.to_owned(),
+                _key1: key1.to_owned(),
             }),
             _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum JpmorganPaymentStatus {
-    Succeeded,
-    Failed,
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionStatus {
+    Success,
+    Denied,
+    Error,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionState {
+    Closed,
+    Authorized,
+    Voided,
     #[default]
-    Processing,
+    Pending,
+    Declined,
+    Error,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganPaymentsResponse {
+    transaction_id: String,
+    request_id: Option<String>,
+    transaction_state: JpmorganTransactionState,
+    response_status: String,
+    response_code: String,
```

Comment:
Can't we use an enum instead here?

Reply:
No, its defined as a `Short explanation for response status` in the docs